presto-common/src/main/java/com/facebook/presto/common/block/RunLengthEncodedBlock.java

```diff
@@ -0,0 +1,414 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.common.block;
+
+import com.facebook.presto.common.predicate.Utils;
+import com.facebook.presto.common.type.Type;
+import io.airlift.slice.Slice;
+import io.airlift.slice.SliceOutput;
+import org.openjdk.jol.info.ClassLayout;
+
+import java.util.function.BiConsumer;
+
+import static com.facebook.presto.common.block.BlockUtil.checkArrayRange;
+import static com.facebook.presto.common.block.BlockUtil.checkValidPosition;
+import static com.facebook.presto.common.block.BlockUtil.checkValidRegion;
+import static com.facebook.presto.common.block.BlockUtil.internalPositionInRange;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class RunLengthEncodedBlock
+        implements Block
+{
+    private static final int INSTANCE_SIZE = ClassLayout.parseClass(RunLengthEncodedBlock.class).instanceSize();
+
+    public static Block create(Type type, Object value, int positionCount)
+    {
+        Block block = Utils.nativeValueToBlock(type, value);
+        if (block instanceof RunLengthEncodedBlock) {
+            block = ((RunLengthEncodedBlock) block).getValue();
+        }
+        return new RunLengthEncodedBlock(block, positionCount);
+    }
+
+    private final Block value;
+    private final int positionCount;
+
+    public RunLengthEncodedBlock(Block value, int positionCount)
+    {
+        requireNonNull(value, "value is null");
+        if (value.getPositionCount() != 1) {
+            throw new IllegalArgumentException(format("Expected value to contain a single position but has %s positions", value.getPositionCount()));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: &#39;getPositionCount&#39; expression. This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 52. Consider formatting this data with &#39;%d&#39; instead.